server/src/resolvers/getVariantsResolver/utils/fetchCaddAnnotations.ts

```diff
@@ -1,86 +1,88 @@
-  cDNApos: 'cdna',
-  protPos: 'aaPos',
-};
+  const tbiIndexed = new TabixIndexedFile({
+    filehandle: new RemoteFile(annotationUrl, { fetch: fetch as Fetcher }),
+    tbiPath: indexPath,
+  });
 
-const _getAnnotations = async (position: string, assemblyId: string) => {
-  const query = await _buildQuery(position, assemblyId);
-  const execPromise = promisify(exec);
-  return execPromise(query, { maxBuffer: 10e7 }); // 100mb
+  const { chromosome, start, end } = resolveChromosome(position);
+
+  const lines: string[] = [];
+  if (chromosome && start && end) {
+    await tbiIndexed.getLines(`${chromosome}`, Number(start) - 1, Number(end) + 1, line => {
+      lines.push(line);
+    });
+  }
+
+  console.log(lines);
+  return lines;
 };
 
-const _formatAnnotations = (annotations: string) => {
-  const annotationsArray = annotations.split('\n');
-  const headerRow = annotationsArray.shift();
+const _formatAnnotations = (annotations: string[]) => {
+  /*
+    indexes for headers in annotation tsv: 
+    https://cadd.gs.washington.edu/static/ReleaseNotes_CADD_v1.6.pdf
+    Chrom: 1
+    Pos: 2
+    Ref: 3
+    Alt: 4
+    Consequence: 8
+    oAA: 17
+    nAA: 18
+    FeatureID: 20
+    cDNApos: 25
+    protpos: 29
 
-  if (!headerRow) {
-    return [];
-  }
+    Note that in JSON, these indexes would start at 0.
+  */
 
-  const headers = headerRow
-    .split(/\W+/)
-    .filter(Boolean)
-    .reduce<Record<number, keyof CaddAnnotation>>(
-      (acc, curr, i) => ({
-        ...acc,
-        [i]: NAME_MAP[curr],
-      }),
-      {}
-    );
+  const headersMap: Record<number, keyof CaddAnnotation> = {
+    0: 'chrom',
+    1: 'pos',
+    2: 'ref',
+    3: 'alt',
+    7: 'consequence',
+    16: 'aaRef',
+    17: 'aaAlt',
+    19: 'transcript',
+    24: 'cdna',
+    28: 'aaPos',
+  };
 
-  return annotationsArray.map(annotation =>
-    annotation
-      .split(/\W+/)
-      .reduce<CaddAnnotation>(
-        (acc, curr, i) => ({ ...acc, [headers[i]]: curr }),
-        {} as CaddAnnotation
-      )
-  );
-};
+  const headersIndex = Object.keys(headersMap);
+  const headers = Object.values(headersMap);
 
-const _buildQuery = async (position: string, assemblyId: string) => {
-  const resolvedAssemblyId = resolveAssembly(assemblyId);
-  const annotationUrl = resolvedAssemblyId === '38' ? ANNOTATION_URL_38 : ANNOTATION_URL_37;
-  const indexPath = resolvedAssemblyId === '38' ? INDEX_38_PATH : INDEX_37_PATH;
+  const result = annotations.map(annotation => {
+    // Get only the required annotation columns
+    const annotationColumns = annotation
+      .split('\t')
+      .filter((a, i) => headersIndex.some(headerId => Number(headerId) === i));
+    return Object.fromEntries(
+      headers.map((h, i) => [h, annotationColumns[i]])
+    ) as unknown as CaddAnnotation;
+  });
 
-  return `tabix -h  ${annotationUrl} ${indexPath} ${position} | awk 'NR!=1{print $1,$2,$3,$4,$8,$17,$18,$20,$25,$29}'`;
+  console.log(result);
+  return result;
```

Comment:
Can you explain what this function is supposed to do?

Reply:
The tabix JS wrapper returns a list of strings, each string represents a line in the `ANNOTATION_URL`, i.e. a variant annotation. It looks something like this: 

`19\t12312321\t23432\t....`

We know what headers correspond to what values based on CADD's documentation [here](https://cadd.gs.washington.edu/static/ReleaseNotes_CADD_v1.6.pdf).

The goal of the function is to take in a list of strings and return a list of objects of type `CaddAnnotation`.

Reply:
I will add more detailed comments to the code.